lntest/itest/lnd_test.go

```diff
@@ -14151,37 +14151,53 @@ func TestLightningNetworkDaemon(t *testing.T) {
 	aliceBobArgs := []string{
 		"--default-remote-max-htlcs=483",
 	}
-	if err = lndHarness.SetUp(aliceBobArgs); err != nil {
-		ht.Fatalf("unable to set up test lightning network: %v", err)
-	}
 
 	t.Logf("Running %v integration tests", len(testsCases))
 	for _, testCase := range testsCases {
-		logLine := fmt.Sprintf("STARTING ============ %v ============\n",
-			testCase.name)
+		testCase := testCase
 
-		err := lndHarness.EnsureConnected(
-			context.Background(), lndHarness.Alice, lndHarness.Bob,
-		)
-		require.NoError(t, err, "unable to connect alice to bob")
+		runTest := func() bool {
+			cleanTestCaseName := strings.Replace(
+				testCase.name, " ", "_", -1,
+			)
 
-		err = lndHarness.Alice.AddToLog(logLine)
-		require.NoError(t, err, "unable to add to Alice's log")
+			err = lndHarness.SetUp(cleanTestCaseName, aliceBobArgs)
+			if err != nil {
+				ht.Fatalf("unable to set up test lightning "+
+					"network: %v", err)
+			}
+			defer func() {
```

Comment:
This now causes to create a log file for each test case, even if I target a specific one (`make itest icase=REST_API` for example) and takes more than 3 minutes. I think we should move all this setup and teardown code inside of the `t.Run()` closure, now that we actually setup and tear down for each test individually.

Reply:
I don't know how I didn't catch this, thank you! :pray: Fixed.